jetty-client/src/main/java/org/eclipse/jetty/client/HttpClient.java

```diff
@@ -427,7 +427,7 @@ public Request newRequest(String uri)
      */
     public Request newRequest(URI uri)
     {
-        return newHttpRequest(newConversation(), uri);
+        return newHttpRequest(newConversation(), checkHostInUri(uri));
```

Comment:
Yes, overlooked it since first modified `newRequest(URI)` and `copyRequest(HttpRequest, URI)`. Will be fixed.